packages/datadog-instrumentations/src/kafkajs.js

```diff
@@ -0,0 +1,114 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const producerStartCh = channel('apm:kafkajs:produce:start')
+const producerMessageCh = channel(`apm:kafkajs:produce:message`)
+const producerFinishCh = channel('apm:kafkajs:produce:finish')
+const producerErrorCh = channel('apm:kafkajs:produce:error')
+
+const consumerStartCh = channel('apm:kafkajs:consume:start')
+const consumerFinishCh = channel('apm:kafkajs:consume:finish')
+const consumerErrorCh = channel('apm:kafkajs:consume:error')
+
+addHook({ name: 'kafkajs', versions: ['>=1.4'] }, (obj) => {
+  const Kafka = obj.Kafka
+  shimmer.wrap(Kafka.prototype, 'producer', createProducer => function () {
+    if (!producerStartCh.hasSubscribers) {
+      return createProducer.apply(this, arguments)
+    }
+    const producer = createProducer.apply(this, arguments)
+    const send = producer.send
+
+    const outerAsyncResource = new AsyncResource('bound-anonymous-fn')
+    const innerAsyncResource = new AsyncResource('bound-anonymous-fn')
+
+    producer.send = function () {
+      return innerAsyncResource.runInAsyncScope(() => {
+        producerStartCh.publish(undefined)
+        try {
+          const lastArgId = arguments.length - 1
+          const cb = arguments[lastArgId]
+          if (typeof cb === 'function') {
```

Comment:
What about promises?

Reply:
That could mean we're missing a test, especially since the previous code was using `tracer.wrap` so it handled promises and callbacks automatically.

Reply:
your right, it is a promise even when looking at the github code it returns a promise. I guess we are missing a test